cmd/mutate-image/main_test.go

```diff
@@ -0,0 +1,307 @@
+		auth := authn.FromConfig(authn.AuthConfig{
+			Username: os.Getenv(regUser),
+			Password: os.Getenv(regPass),
+		})
+
+		ref, err := name.ParseReference(image)
+		Expect(err).To(BeNil())
+
+		img, err := remote.Image(ref, remote.WithAuth(auth))
+		Expect(err).To(BeNil())
+
+		manifest, err := img.Manifest()
+		Expect(err).To(BeNil())
+
+		return manifest.Annotations[annotation]
+	}
+
+	withTempFile := func(pattern string, f func(filename string)) {
+		file, err := ioutil.TempFile(os.TempDir(), pattern)
+		Expect(err).ToNot(HaveOccurred())
+		defer os.Remove(file.Name())
+
+		f(file.Name())
+	}
+
+	filecontent := func(path string) string {
+		data, err := ioutil.ReadFile(path)
+		Expect(err).ToNot(HaveOccurred())
+		return string(data)
+	}
+
+	getImage := func(tag name.Tag) v1.Image {
+		auth := authn.FromConfig(authn.AuthConfig{
+			Username: os.Getenv(regUser),
+			Password: os.Getenv(regPass),
+		})
+
+		ref, err := name.ParseReference(tag.String())
+		Expect(err).To(BeNil())
+
+		desc, err := remote.Get(ref, remote.WithAuth(auth))
+		Expect(err).To(BeNil())
+
+		img, err := desc.Image()
+		Expect(err).To(BeNil())
+
+		return img
+	}
+
+	getImageDigest := func(tag name.Tag) v1.Hash {
+		digest, err := getImage(tag).Digest()
+		Expect(err).To(BeNil())
+
+		return digest
+	}
+
+	Context("validations and error cases", func() {
+		It("should fail in case mandatory arguments are missing", func() {
+			Expect(run()).To(HaveOccurred())
+		})
+
+		It("should fail in case --image is empty", func() {
+			Expect(run("--image", "")).To(HaveOccurred())
+		})
+
+		It("should fail in case --image does not exist", func() {
+			Expect(run(
+				"--image",
+				"docker.io/feqlQoDIHc/bcfHFHHXYF",
+			)).To(HaveOccurred())
+		})
+
+		It("should fail in case annotation is invalid", func() {
+			tag := pushImage("test1")
+
+			Expect(run(
+				"--image",
+				tag.String(),
+				"--annotation",
+				"org.opencontainers.image.url*https://my-company.com/images",
+			)).To(HaveOccurred())
+		})
+
+		It("should fail in case label is invalid", func() {
+			tag := pushImage("test2")
+
+			Expect(run(
+				"--image",
+				tag.String(),
+				"--label",
+				" description*image description",
+			)).To(HaveOccurred())
+		})
+	})
+
+	Context("mutating the image", func() {
+		It("should mutate an image with single annotation", func() {
+			tag := pushImage("test3")
+
+			Expect(run(
+				"--image",
+				tag.String(),
+				"--annotation",
+				"org.opencontainers.image.url=https://my-company.com/images",
+			)).To(BeNil())
+
+			Expect(getImageAnnotation(tag.String(), "org.opencontainers.image.url")).
+				To(Equal("https://my-company.com/images"))
+		})
+
+		It("should mutate an image with multiple annotations", func() {
+			tag := pushImage("test4")
+
+			Expect(run(
+				"--image",
+				tag.String(),
+				"--annotation",
+				"org.opencontainers.image.url=https://my-company.com/images,org.opencontainers.image.source=https://github.com/org/repo",
```

Comment:
As mentioned previously, I think that a comma is a valid character inside an annotation value. As such we must support multiple args and commas inside tha value must stay part of the value. Please extend and adjust the test suite accordingly.